scripts/gb-model/create_region_shapes.py

```diff
@@ -0,0 +1,421 @@
+# SPDX-FileCopyrightText:  Open Energy Transition gGmbH
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+
+"""
+Script to create region shapes by dividing country shapes based on ETYS boundary lines.
+
+This script reads country shapes from a GeoJSON file and divides them using
+boundary lines from the ETYS boundary data to create regional divisions.
+The resulting regions are saved as a new GeoJSON file.
+"""
+
+import logging
+import os
+from pathlib import Path
+
+import geopandas as gpd
+import pandas as pd
+from shapely.ops import split
+from shapely.geometry import Point
+
+from scripts._helpers import configure_logging, set_scenario_config
+
+logger = logging.getLogger(__name__)
+
+
+def load_country_shapes(filepath):
+    """
+    Load country shapes from GeoJSON file.
+
+    Args:
+        filepath (str): Path to the country shapes GeoJSON file
+
+    Returns:
+        geopandas.GeoDataFrame: Country shapes data
+    """
+    try:
+        logger.info(f"Loading country shapes from: {filepath}")
+        country_gdf = gpd.read_file(filepath)
+        logger.info(f"Loaded {len(country_gdf)} country shapes")
+        return country_gdf
+    except Exception as e:
+        logger.error(f"Error loading country shapes: {e}")
+        raise
+
+
+def load_boundary_lines(filepath):
+    """
+    Load boundary lines from shapefile or GeoJSON.
+
+    Args:
+        filepath (str): Path to the boundary lines file
+
+    Returns:
+        geopandas.GeoDataFrame: Boundary lines data
+    """
+    try:
+        logger.info(f"Loading boundary lines from: {filepath}")
+        boundary_gdf = gpd.read_file(filepath)
+        logger.info(f"Loaded {len(boundary_gdf)} boundary features")
+        return boundary_gdf
+    except Exception as e:
+        logger.error(f"Error loading boundary lines: {e}")
+        raise
```

Comment:
remove the try-except and rather let the workflow fail with a clear error message